ivanti_security_controls/icon_ivanti_security_controls/actions/start_patch_scan/action.py

```diff
@@ -39,19 +38,25 @@ def run(self, params={}):
             "templateId": params.get(Input.TEMPLATE_ID),
             "useMachineCredential": use_machine_credential
         }
-        scan_details = self.connection.ivanti_api.start_patch_scan(payload)
-
-        i = 0
-        # Poll for patch scan completion
-        while i < max_poll_time:
-            time.sleep(10)
-            patch_scan_status_details = self.connection.ivanti_api.get_patch_scan_status_details(scan_details['id'], allow_404=True)
-            if patch_scan_status_details is not None:
-                if patch_scan_status_details.get('isComplete'):
-                    scan_details['isComplete'] = patch_scan_status_details['isComplete']
-                    scan_details['updatedOn'] = patch_scan_status_details['updatedOn']
-                    break
-            i += 10
+        
+        self.connection.ivanti_api.create_session_credential()
+
+        scan = self.connection.ivanti_api.start_patch_scan(payload)
+
+        operation_location_url = scan.headers.get("Operation-Location")
+
+        try:
+            polling2.poll(lambda: self.connection.ivanti_api.get_operation_location(operation_location_url)
+            .get("percentComplete") == 100, step=10, timeout=max_poll_time)
```

Comment:
Log for `KeyError` if `percentComplete` isn't returned

Reply:
@zyoutz-r7 I've noticed that this operation-location URL doesn't always contain percentComplete straight away. It happens after around 20-40 seconds, that's why I'm not logging this exception.